tpm2/Makefile

```diff
@@ -73,16 +80,33 @@ UTIL_SOURCES = $(wildcard $(UTIL_SRC_DIR)/*c)
 UTIL_OBJ_DIR = $(OBJ_DIR)/util
 UTIL_OBJECTS = $(subst $(UTIL_SRC_DIR), $(UTIL_OBJ_DIR), $(UTIL_SOURCES:%.c=%.o))
 
+# Specify logger source/object/header files
+LOGGER_SOURCES = $(wildcard $(LOGGER_SRC_DIR)/*c)
+LOGGER_OBJECTS = $(subst $(LOGGER_SRC_DIR), $(LOGGER_OBJ_DIR), $(LOGGER_SOURCES:%.c=%.o))
+
 # Specify Kmyth header files
 HEADER_FILES = $(wildcard $(INCLUDE_DIR)/*h)
+LOGGER_HEADER_FILES = $(wildcard $(LOGGER_INC_DIR)/*h)
+
+#Specify path for logger output of shared library
+LOGGER_LIB_NAME = kmythlogger
+LOGGER_LIB_SONAME = lib$(LOGGER_LIB_NAME).so
+LOGGER_LIB_LOCAL_DEST = $(LOGGER_LIB_DIR)/$(LOGGER_LIB_SONAME)
+LOGGER_LIB_INSTALL_DIR = $(PREFIX)/lib/
 
-# Consolidate object directories to simplify mkdir call for 'pre' target below
+# Consolidate created directories to simplify mkdir call for 'pre' target below
 OBJECT_DIRS = $(SEAL_OBJ_DIR)
 OBJECT_DIRS += $(UNSEAL_OBJ_DIR)
 OBJECT_DIRS += $(GETKEY_OBJ_DIR)
 OBJECT_DIRS += $(UTIL_OBJ_DIR)
+OBJECT_DIRS += $(LOGGER_OBJ_DIR)
+DOC_DIRS = $(DOC_DIR)
+DOC_DIRS += $(LOGGER_DOC_DIR)
 
-all: pre kmyth-seal kmyth-unseal kmyth-getkey
+all: pre logger kmyth-seal kmyth-unseal kmyth-getkey
```

Comment:
Since 'all' is not a file-based target, I think it should be phony (.PHONY: all).

Reply:
I added .PHONY to all, pre, and doc